exercise/submission_models.py

```diff
@@ -36,6 +36,8 @@ def create_from_post(self, exercise, submitters, request):
         ]
         try:
             meta_data_dict = json.loads(request.POST.get('__aplus__', '{}'))
+            if 'lang' not in meta_data_dict:
```

Comment:
Typically, the try-except shouldn't include only as little code as possible. In other words, this part could be moved out of the block.

As a side note, try-except-else is nice pattern for this kind of things, specially when except doesn't raise/return. Here that wouldn't be so useful.